dataduct/database/select_statement.py

```diff
@@ -16,8 +16,8 @@ def __init__(self, sql):
         """
         super(SelectStatement, self).__init__(sql)
 
-        self._dependencies = parse_select_dependencies(self.sql())
-        self._raw_columns = parse_select_columns(self.sql())
+        self._dependencies = parse_select_dependencies(self)
```

Comment:
change the functions in https://github.com/coursera/dataduct/blob/develop/dataduct/database/parsers/select_query.py to assume they are getting string instead of this as that is what we do for views and tables.